codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/EnumGenerator.kt

```diff
@@ -189,14 +197,16 @@ class EnumGenerator(
                 }
             }
         }
+    }
 
+    private fun renderFromStr() {
         writer.rust(
             """
             impl std::str::FromStr for $enumName {
                 type Err = std::convert::Infallible;
 
                 fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
-                    Ok($enumName::from(s))
+                    Ok($enumName::try_from(s).unwrap())
```

Comment:
`EnumGenerator` should use `from`, not `try_from`. This will cause the client to panic.